airflow/jobs.py

```diff
@@ -378,13 +378,24 @@ def import_errors(self, dagbag):
                 filename=filename, stacktrace=stacktrace))
         session.commit()
 
+    def _normalize_schedule(self, dag, next_run_date, dttm):
+        if dag.previous_schedule(next_run_date) == dttm:
```

Comment:
Should that be DAG method alongside `following_schedule` and `previous_schedule`?
```
#in DAG
def normalize_schedule(self, dag, dttm):
    # based on the idea that going to previous and back to next should land you where you started
    following = self.following_schedule(dttm)
    if self.previous_schedule(next) != dttm:
        return following
    return dttm
```